services/manage.py

```diff
@@ -5,30 +5,33 @@
 
 
 from models.stream import Stream
+from models.connexus_user import ConnexusUser
 
 class Manage(webapp2.RequestHandler):
 
     def get(self):
         self.response.headers['Content-Type'] = 'application/json'
-        user = self.request.get('user')
+        user_id = self.request.get('user')
         stream_type = self.request.get('type')
         if stream_type == 'own' :
-            streams = Stream.query(Stream.owner == user)
+            streams = Stream.query(Stream.owner == user_id)
         else:
-            #TODO : handle suscription views
-            streams = [] 
-        
+            user = ConnexusUser.from_user_id(user_id)
+            streams = user.get_subscribed_streams()
+
         stream_list = []
 
         if streams is not None:
             for stream in streams:
-                date = stream.date
-                date_string = date.strftime('%m/%d/%Y')
-                stream_list.append({
-                'id': stream.key.id(), 
-                'name': stream.name, 
-                'date': date_string, 
-                'photo_count': len(stream.photos)
-                })
+                if stream is not None:
+                    date = stream.date
+                    date_string = date.strftime('%m/%d/%Y')
+                    stream_list.append({
+                    'id': stream.key.id(), 
+                    'name': stream.name, 
+                    'date': date_string, 
+                    'photo_count': len(stream.photos),
+                    'all_time_views': len(stream.views_list)
```

Comment:
All time views should not be the views_list since entries in this list are purged every hour with the cron. There is another field in the model called "views" which reflects the all time views. 

Crons don't run locally, you have to manually run it to test it, but if you do, you'll notice most of the items in the view_list will be removed.

Reply:
And this is why we have code reviews.

Reply:
I LOVE this feature. I am so sad I have no one at work to do this with, haha